src/Shared/ParameterBindingMethodCache.cs

```diff
@@ -126,7 +130,7 @@ public bool HasTryParseMethod(Type type)
                     expression);
             }
 
-            methodInfo = GetStaticMethodFromHierarchy(type, "TryParse", new[] { typeof(string), typeof(IFormatProvider), type.MakeByRefType() }, ValidateReturnType);
+            methodInfo = GetStaticMethodFromHierarchy(type, "TryParse", new[] { typeof(string), typeof(IFormatProvider), type.MakeByRefType() }, ValidateReturnType, _throwOnInvalidMethod);
```

Comment:
Can we make these methods instance methods instead of statics so we can access the property on the class instead of piping it through?

I sense a bit of a code smell with passing the magic boolean around.